components/BurgerMenuNavCategories.vue

```diff
@@ -0,0 +1,342 @@
+              icon="angle-left"
+            />
+            {{ item.name }}
+            <fa
+              v-if="!isCategoryOpen(index)"
+              class="burger-menu-nav-list-item-category__come-back"
+              icon="angle-right"
+            />
+          </button>
+        </div>
+        <div v-show="isCategoryOpen(index)">
+          <burger-menu-nav-sections
+            :sections="item.sections"
+            @close-menu="handleLinkClick"
+          />
+        </div>
+      </div>
+      <div v-else v-show="openDropdownIndex === -1" @click="handleLinkClick">
+        <pix-prismic-link
+          :field="item"
+          class="burger-menu-nav-list-item__link"
+          :class="{ 'blue-link': shouldApplyBlueClass(item.url) }"
+        >
+          <img
+            v-if="item.beforeIcon"
+            :src="`/images/${item.beforeIcon}`"
+            :alt="item.alternativeTextForBeforeIcon"
+            role="presentation"
+            width="22"
+            height="18"
+          />
+          {{ item.name }}
+          <img
+            v-if="item.afterIcon"
+            :src="`/images/${item.afterIcon}`"
+            :alt="item.alternativeTextForAfterIcon"
+            role="presentation"
+            width="22"
+            height="18"
+          />
+        </pix-prismic-link>
+      </div>
+    </li>
+  </ul>
+</template>
+
+<script>
+import BurgerMenuNavSections from '~/components/BurgerMenuNavSections.vue'
+
+export default {
+  name: 'BurgerMenuNavCategories',
+  components: {
+    BurgerMenuNavSections,
+  },
+  props: {
+    items: {
+      type: Array,
+      default: null,
+    },
+    isLocaleSwitcherOpen: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      openDropdownIndex: -1,
+    }
+  },
+  computed: {
+    isLocaleSwitcherOpenLocal: {
+      get() {
+        return this.isLocaleSwitcherOpen
+      },
+      set(value) {
+        if (value) {
+          this.openDropdownIndex = -1
+        }
+      },
+    },
+    navigationLinks() {
+      if (this.items.length === 0) {
+        return null
+      }
+
+      return this.items.reduce((formattedData, currentItem) => {
+        const isCategory =
+          currentItem.categoryName.length > 0 &&
+          currentItem.categoryName[0].text
+
+        if (!isCategory) {
+          formattedData.push({
+            name: this.getTextFromArray(currentItem.menuName),
+            beforeIcon: currentItem.beforeIcon,
+            alternativeTextForBeforeIcon: currentItem
+              .alternativeTextForBeforeIcon[0]
+              ? currentItem.alternativeTextForBeforeIcon[0].text
+              : '',
+            afterIcon: currentItem.afterIcon,
+            alternativeTextForAfterIcon: currentItem
+              .alternativeTextForAfterIcon[0]
+              ? currentItem.alternativeTextForAfterIcon[0].text
+              : '',
+            url: currentItem.menuLink.url,
+            sections: [],
+          })
```

Comment:
```suggestion
          const firstLevelLink = createFirstLevelLink(currentItem);
          
          formattedData.push(firstLevelLink)
```

Comme pour `createSection` et `createCategory`, utiliser une méthode `createFirstLevelLink` pour alléger la computed ?